### PR TITLE
Check for update

### PR DESCRIPTION
**Pull request by alex-oreshko**
CheckForUpdate

### CODE REVIEW

File: AppCenterAssets/AppCenterAssets/Internals/Manager/MSAssetsUpdateManager.h [hunk 1]
```diff
@@ -6,13 +6,9 @@
...
 
 
-/**
- * Gets current package json object.
- *
- * @return current package json object.
- */
-- (MSLocalPackage *)getCurrentPackage:(NSError **)error;
 
+- (MSLocalPackage *)getCurrentPackage:(NSError **)error;
```
Comment (max-mironov) [inline AppCenterAssets/AppCenterAssets/Internals/Manager/MSAssetsUpdateManager.h]: @alex-oreshko I think would be good to also add comments in h files (we could take them mostly from Android part I believe)

File: Puppet/Puppet/MSAssetsViewController.m [hunk 2]
```diff
@@ -19,7 +20,13 @@ - (void)viewDidLoad {
     self.enabled.on = [MSAssets isEnabled];
     
     _assetsDeployment = [MSAssets makeDeploymentInstanceWithBuilder:^(MSAssetsBuilder *builder) {
-        [builder setDeploymentKey:@"123"];
+
+        NSDictionary *infoDictionary = [[NSBundle mainBundle] infoDictionary];
+        NSString *deploymentKey = [infoDictionary objectForKey:@"MSAssetsDeploymentKey"];
+        [builder setDeploymentKey:deploymentKey];
+        NSString *serverUrl = [infoDictionary objectForKey:@"MSAssetsServerUrl"];
```
Comment (max-mironov) [inline Puppet/Puppet/MSAssetsViewController.m]: We can omit this setting I think (as it should be the default one)  unless we do not want to use custom serverUrl

File: AppCenterAssets/AppCenterAssets/MSAssetsDeploymentInstance.m [hunk 3]
```diff
@@ -101,9 +112,40 @@ - (MSLocalPackage *)getUpdateMetadataForState:(MSAssetsUpdateState)updateState
...
+    BOOL currentUpdateIsPending = [MSAssetsSettingManager isPendingUpdate:package.packageHash];
+
+    if (updateState == MSAssetsUpdateStatePending && !currentUpdateIsPending) {
+        // The caller wanted a pending update
+        // but there isn't currently one.
+        return nil;
+    } else if (updateState == MSAssetsUpdateStateRunning && currentUpdateIsPending) {
+        // The caller wants the running update, but the current
+        // one is pending, so we need to grab the previous.
+        return [[[self managers] updateManager] getPreviousPackage:error];
```
Comment (annakocheshkova) [inline AppCenterAssets/AppCenterAssets/MSAssetsDeploymentInstance.m]: `&error` ?

File: AppCenterAssets/AppCenterAssets/MSAssetsManagers.h [hunk 4]
```diff
@@ -1,10 +1,13 @@
...
 #import "MSAssetsAcquisitionManager.h"
+#import "MSAssetsSettingManager.h"
 
 @interface MSAssetsManagers : NSObject
 
 @property (nonatomic, copy, readonly) MSAssetsUpdateManager *updateManager;
 
 @property (nonatomic, copy, readonly) MSAssetsAcquisitionManager *acquisitionManager;
 
+//@property (nonatomic, copy, readonly) MSAssetsSettingManager *settingManager;
```
Comment (annakocheshkova) [inline AppCenterAssets/AppCenterAssets/MSAssetsManagers.h]: No need - it is static

File: AppCenterAssets/AppCenterAssets/Internals/Manager/MSAssetsUpdateManager.h (hunk 1 above)
Comment (alex-oreshko) [reply AppCenterAssets/AppCenterAssets/Internals/Manager/MSAssetsUpdateManager.h]: Ok

File: Puppet/Puppet/MSAssetsViewController.m (hunk 2 above)
Comment (alex-oreshko) [reply Puppet/Puppet/MSAssetsViewController.m]: If I omit server setting Puppet, everything fails

**Pull request merged**